### ISSUE TITLE
The float converter does not recognise negative numbers

### ISSUE THREAD

**Issue by robinhouston** [pallets/werkzeug#729]
I’m writing an application where the URLs contain floating point numbers that represent latitudes and longitudes. I was surprised to find I could not use the werkzeug.routing.FloatConverter for this, since it does not recognise negative numbers.

**Comment by untitaker** [comment]
That's intentional, but I don't see a reason for this behavior documented anywhere.

**Comment by idwright** [comment]
I'm also hitting this problem.
Is there any reason that -? couldn't be added to the front of the regex?

(I won't even mention problems with using float instead of Decimal....)

**Comment by davidism** [comment]
If someone wants this, make a pull request for it.

**Comment by untitaker** [comment]
This needs to be discussed first, since it is backwards-incompatible.
@mitsuhiko's decision.

**Comment by RonnyPfannschmidt** [comment]
the introduction of a new signedfloat converter would be backward compatible

**Comment by davidism** [comment]
I don't find the backwards incompatibility particularly significant here. Add a note to the changelog "The `int` and `float` converters now recognizes negative values. Pass `signed=False` to keep the previous behavior and only match positive values."

**Comment by RonnyPfannschmidt** [comment]
minor details are significant for those whose code they break ^^
the default at introduction time should be none and a warning that around a certain point in time the default will be set

**Comment by ThiefMaster** [comment]
Parametrizing converters isn't even that straightforward, so I think having the old float and a new signed-float (sfloat?) type would make more sense.

**Comment by fake-name** [comment]
This is also equally true for the integer (`int`) converter, and similarly annoying there, particularly if there's no underlying reason for it to be as such.

**Comment by heavyimage** [comment]
Hey @davidism -- Thanks for the redirect here and sorry to kick a hornet's nest.  I skimmed over the whole issue thread before I posted and it def seemed like there were reasons for why it was impossible to modify the existing converters to also support negative numbers though I'll confess I wasn't sure what they were.  

In terms of moving forward constructively, what I'm (still!) happy to do is create a pull request defining two new converters (one for floats and one for ints) that use different negative-number-supporting regexes.  That seems to be a nice way to preserve the legacy functionality and help out new users of Werkzeug, no?
